packages/system/src/style.js

```diff
@@ -171,6 +171,21 @@ function indexGeneratorsByProp(styles) {
   return index
 }
 
+function getMediaOrder(styles, props) {
+  const medias = {} 
+  const breakpoints = Object.values(getBreakpoints(props))
+
+  for (const breakpoint in breakpoints) {
+    const currentMediaKey = `@media (min-width: ${breakpoints[breakpoint]}px)`;
+    const isValid = Object.keys(styles).includes(currentMediaKey);
```

Comment:
Please extract `Object.keys(styles)` out of the `for` to avoid doing it in each iteration.